packages/webapp/src/components/Animals/AddAnimalsDetails/type.ts

```diff
@@ -0,0 +1,100 @@
+/*
+ *  Copyright 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import { UseFormReturn } from 'react-hook-form';
+import { TFunction } from 'react-i18next';
+
+export type ReactSelectOption<T extends string | number> = {
+  label: string;
+  value: T;
+};
+
+export enum DetailsFields {
+  // GENERAL
+  NAME = 'name',
+  TYPE = 'type',
+  BREED = 'breed',
+  SEX = 'sex',
+  USED_FOR_PRODUCTION = 'used_for_production',
```

Comment:
This field is meant to be used for REproduction, correct? It's not a separate field?